server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/RebalanceOnlyWhenActiveAllocationDecider.java

```diff
@@ -18,11 +18,21 @@ public class RebalanceOnlyWhenActiveAllocationDecider extends AllocationDecider
 
     public static final String NAME = "rebalance_only_when_active";
 
+    static final Decision YES_ALL_REPLICAS_ACTIVE = Decision.single(
+        Decision.Type.YES,
+        NAME,
+        "rebalancing is allowed as all replicas are active in the cluster"
```

Comment:
This is a per-shard check, but "in the cluster" suggests it's looking at all the shards in the cluster.